src/main/java/xyz/wagyourtail/jsmacros/client/api/classes/PlayerInput.java

```diff
@@ -0,0 +1,284 @@
+     * Parses a JSON string into a {@code PlayerInput} Object
+     * For details see {@code PlayerInput.fromCsv()}, on what has to be present.<br>
+     * Capitalization of the keys doesn't matter.
+     *
+     * @param json JSON string to be parsed
+     * @return The JSON parsed into a {@code PlayerInput}
+     * @see #fromCsv(String, String)
+     * @since 1.4.x
+     */
+    public static PlayerInput fromJson(String json) {
+        Map<String, String> map = new HashMap<>();
+        map = (Map<String, String>) new Gson().fromJson(json, map.getClass());
+        map = map.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().toLowerCase().replace("\"", ""), e -> e.getValue().replace("\"", "")));
+        return fromMap(map);
+    }
+
+
+    /**
+     * Converts a Map, which has all keys lowercase and that contains all the prerequisites listed in {@code PlayerInput.fromCSV}.
+     *
+     * @param input Map to be converted
+     * @return The map converted into a {@code PlayerInput}
+     * @see #fromCsv(String, String)
+     * @since 1.4.x
+     */
+    private static PlayerInput fromMap(Map<String, String> input) {
+        return new PlayerInput(
+                input.get("movementforward") != null ? Float.parseFloat(input.get("movementforward")) : input.get("w").equals(input.get("s")) ? 0.0F : (Boolean.parseBoolean(input.get("w")) ? 1.0F : -1.0F),
+                input.get("movementsideways") != null ? Float.parseFloat(input.get("movementsideways")) : input.get("a").equals(input.get("d")) ? 0.0F : (Boolean.parseBoolean(input.get("a")) ? 1.0F : -1.0F),
+                input.get("yaw") != null ? Float.parseFloat(input.get("yaw")) : 0.0f,
+                input.get("pitch") != null ? Float.parseFloat(input.get("pitch")) : 0.0f,
+                input.get("jumping") != null ? Boolean.parseBoolean(input.get("jumping")) : input.get("jump") != null && Boolean.parseBoolean(input.get("jump")),
+                input.get("sneaking") != null ? Boolean.parseBoolean(input.get("sneaking")) : input.get("sneak") != null && Boolean.parseBoolean(input.get("sneak")),
+                input.get("sprinting") != null ? Boolean.parseBoolean(input.get("sprinting")) : input.get("sprint") != null && Boolean.parseBoolean(input.get("sprint"))
+        );
+    }
+
+    /**
+     * Converts the current object into a string.
+     * This can be used to convert current inputs created using {@code Player.getCurrentPlayerInput()}
+     * to either JSON or CSV.
+     * <p>
+     * The output can be converted into a PlayerInput object again by using either
+     * {@code fromCsv(String, String)} or {@code fromJson(String)}.
+     *
+     * @param wasd     whether to convert {@code movementForward/Sideways} into {@code WASD} values
+     * @param varNames whether to include variable Names(=JSON) or not(=CSV)
+     * @return The {@code PlayerInput} object as a string
+     * @since 1.4.x
+     */
+    public String toString(boolean wasd, boolean varNames) {
```

Comment:
with a `fromJson` and `fromCsv` I was kinda expecting a `toJson` and `toCsv`

Reply:
Should I create those functions, which then call toString?
But since the only diff is varNames, I don't know

Reply:
I guess it doesnt matter than, you can leave as-is